### PR TITLE
Misc doc changes

### PR DESCRIPTION
**Pull request by kianmeng**
Besides other documentation changes, this commit ensures the generated
HTML doc for HexDocs.pm will become the main source doc for this Elixir
library which leverage on latest ExDoc features.

Screenshot:
Screenshot-20210404154157-2672x1458

### CODE REVIEW

File: .formatter.exs [hunk 1]
```diff
@@ -1,3 +1,4 @@
+# Used by "mix format"
 [
-  inputs: ["*.{ex,exs}", "{config,lib,test}/**/*.{ex,exs}"]
+  inputs: ["*. {ex,exs}", "{config,lib,test}/**/*.{ex,exs}"]
```
Comment (princemaple) [inline .formatter.exs:3]:
```suggestion
  inputs: ["*.{ex,exs}", "{config,lib,test}/**/*.{ex,exs}"]
```

**Review by princemaple — APPROVED** [review]
Thanks :)

**Comment by kianmeng** [comment]
@princemaple Okay if we push the documentation convention (for example, table of contents) to other Elixir libraries / projects under Swoosh? Let me know.

**Comment by princemaple** [comment]
Sure. I actually did some rearrangement of the docs in Swoosh. Any improvement is welcome.

**Pull request merged**